### DISCUSSION TITLE
Trouble Creating Custom Geometry

### DISCUSSION THREAD

**Discussion by CunningEagle** [robbievanleeuwen/section-properties#273] [Q&A] [answered]
Hello all,

I was able to generate meshes of the primitive sections and the example custom geometry without issues, but I tried to create my own custom shape (just a basic 1x1 square) and I'm getting an error (ValueError: Input must have at least three vertices.).  I read over the API documentation and am struggling to diagnose the problem.  Here is my code in Colab:

%pip install sectionproperties
from sectionproperties.pre.geometry import Geometry
from sectionproperties.analysis.section import Section

points = [[0,0], 
          [0,1], 
          [1,1],
          [1,0]]
facets = []
holes = []
control_points = [[0.5,0.5]]

geometry = Geometry.from_points(points, facets, control_points)

geometry = geometry.create_mesh(mesh_sizes=2.5)

section = Section(geometry)
section.plot_mesh()  # plot the generated mesh

section.calculate_geometric_properties()

section.plot_centroids()

Any guidance would be appreciated.  Thanks!

**Comment by CunningEagle** [comment, marked as answer]
I figured out the issue (facets are not optional).  Thanks anyway.

**Reply by robbievanleeuwen** [reply]
Hi @CunningEagle, apologies for the delayed response. Yes this is currently an issue that requires fixing #255. For now the best approach is to supply a facet list - aiming to fix this with the ``v3.0`` release.
